internal/infrastructure/translator.go

```diff
@@ -0,0 +1,46 @@
+package infrastructure
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/envoyproxy/gateway/api/config/v1alpha1"
+	"github.com/envoyproxy/gateway/internal/envoygateway/config"
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+// Translate translates the provided infra ir into managed infrastructure.
+func Translate(ctx context.Context, cfg *config.Server, infra *ir.Infra) (*Manager, error) {
+	if err := ir.ValidateInfra(infra); err != nil {
+		return nil, err
+	}
+
+	if cfg == nil {
+		return nil, errors.New("server config is nil")
+	}
+
+	mgr, err := NewManager(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	log := cfg.Logger
+
+	if cfg.EnvoyGateway == nil ||
+		cfg.EnvoyGateway.Provider == nil ||
+		cfg.EnvoyGateway.Provider.Type == v1alpha1.ProviderTypeKubernetes {
```

Comment:
IMO this would be a good place to implement the protobuf-esque of implementing `GetXXX()` methods that fill defaults and handle nils; so that this check would be `cfg.GetEnvoyGateway().GetProvider().GetType() == v1alpha1.ProviderTypeKubernetes`.